lib/core/context.c

```diff
@@ -187,3 +187,25 @@ void h2o_context_update_timestamp_string_cache(h2o_context_t *ctx)
     h2o_time2str_rfc1123(ctx->_timestamp_cache.value->rfc1123, &gmt);
     h2o_time2str_log(ctx->_timestamp_cache.value->log, ctx->_timestamp_cache.tv_at.tv_sec);
 }
+
+int h2o_context_close_idle_connections(h2o_context_t *ctx, int max_connections_to_close, int min_age)
+{
+    int closed = 0;
+    h2o_linklist_t *node, *nprev;
+    if (max_connections_to_close <= 0)
+        return 0;
+    for (node = ctx->_inactive_conns.prev; node != &ctx->_inactive_conns; node = nprev) {
```

Comment:
Shouldn't we iterate the list in the normal order (i.e. by using `next`)?

I assume that the elements are ordered from the oldest to the newest, because we insert connections like by doing `h2o_linklist_insert(&conn->super.ctx->_inactive_conns, &conn->super._conns)`, which inserts the element at the end of the list. Closing the ones that have spent longer time in idle state makes sense to me.